### ISSUE TITLE
terrform apply bug

### ISSUE THREAD

**Issue by DoctusHartwald** [anubhavmishra/vault-on-azure#3]
hello,
i"ve followed the readme setting up 
az account set --subscription="$SUBSCRIPTION_ID"

which version terraform shall I use ?
**the prompt asking a ssh key ...where do you get it** ?
downgrade on 0.11 version 

./terraform apply
var.ssh_key_private
  Enter a value:

var.ssh_key_private
  Enter a value:

var.ssh_key_private
  Enter a value:

var.ssh_key_private
  Enter a value:

Error: Error asking for user input: missing required value for "ssh_key_private"
what shall i do ????

**Terraform v0.12.26 on AZURE**
There are some problems with the configuration, described below.

The Terraform configuration must be valid before initialization so that
Terraform can determine which modules and providers need to be installed.

Warning: Interpolation-only expressions are deprecated

  on main.tf line 5, in resource "azurerm_resource_group" "default":
   5:   name     = "${var.resource_group_name}"

Terraform 0.11 and earlier required all non-constant expressions to be
provided via interpolation syntax, but this pattern is now deprecated. To
silence this warning, remove the "${ sequence from the start and the }"
sequence from the end of this expression, leaving just the inner expression.

Template interpolation syntax is still used to construct strings from
expressions when the template includes multiple interpolation sequences or a
mixture of literal strings and interpolations. This deprecation applies only
to templates that consist entirely of a single interpolation sequence.

(and 24 more similar warnings elsewhere)

Error: Extraneous label for module

  on main.tf line 13, in module "network" "vault-demo-network":
  13: module "network" "vault-demo-network" {

Only 1 labels (name) are expected for module blocks.

**Comment by anubhavmishra** [comment]
Currently, the project support Terraform version 0.11. The two variables aren't used in the project anymore and I have removed them. You can try to reclone the repo and try again.

**Comment by DoctusHartwald** [comment]
i've just retried on my azure acocunt its failing on the next steps 
`./terraform apply

Error: azurerm_network_interface.vault-demo: : invalid or unknown key: network_security_group_id



Error: azurerm_public_ip.vault-demo: "allocation_method": required field is not set



Error: azurerm_public_ip.vault-demo: : invalid or unknown key: public_ip_address_allocation



Error: azurerm_virtual_machine_extension.vault-demo: "virtual_machine_id": required field is not set



Error: azurerm_virtual_machine_extension.vault-demo: : invalid or unknown key: location



Error: azurerm_virtual_machine_extension.vault-demo: : invalid or unknown key: resource_group_name



Error: azurerm_virtual_machine_extension.vault-demo: : invalid or unknown key: virtual_machine_name



Error: module.network.azurerm_subnet.subnet[0]: : invalid or unknown key: network_security_group_id



Error: module.network.azurerm_subnet.subnet[1]: : invalid or unknown key: network_security_group_id



Error: provider.azurerm: "features": required field is not set`

**Comment by lzc9pk** [comment]
Hello DoctusHartwald,

I am running into the same exact issue. Were you able to resolve the same  and how?
What exact version of Terraform version 0.11 did you use ?

**Comment by DoctusHartwald** [comment]
yes i have used this version but in azure there were some issue while applying the terraform.
I didn't push the solution.. as if its not working I did it in GCP or AWS its working fine . Hope it will be resolve soon on azure
